handlers/handlers.go

```diff
@@ -131,34 +131,37 @@ func HandleSessionRequest(
 
 	switch request.Method {
 	case http.MethodPost:
-		var loginForm LoginForm
-		body := getRequestBody(request)
+		loginForm := new(LoginForm)
 
-		if err := json.Unmarshal(body, &loginForm); err != nil {
-			panic(err)
+		if err := json.NewDecoder(request.Body).Decode(loginForm); err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+			return
 		}
 
 		if err := userservice.AuthenticateUserCredentials(
 			models.NewEmailAddress(loginForm.EmailAddress),
 			loginForm.Password,
 		); err != nil {
-			panic(err)
+			http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+			return
 		}
 
 		// Set our cookie to have a valid JWT Token as the value
 		{
 			userId, err := userservice.GetIdForUserWithEmailAddress(
 				models.NewEmailAddress(loginForm.EmailAddress))
 			if err != nil {
-				panic(err)
+				http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+				return
 			}
 
-			token, err := createTokenAsString(userId, credentialTimeoutInMinutes)
+			token, err := createTokenAsString(userId, credentialTimeoutDuration)
 			if err != nil {
-				panic(err)
+				http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
```

Comment:
this is the sort of thing we'd ideally handle in some sort of middleware (seeing how often you've had to write it), but this is perfectly good for now